j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt

```diff
@@ -328,6 +319,13 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
                 text = text.replace("\\\\([0-3]?[0-7]{1,2})".toRegex()) {
                     String.format("\\u%04x", Integer.parseInt(it.groupValues[1], 8))
                 }
+                val parent = expression.parent
+                if (parent is PsiBinaryExpression) {
+                    val castNeeded = (parent.operationTokenType !in COMPARISON_OPS) || !isChar(parent.rOperand)
+                    if (castNeeded) {
+                        text += ".${Char::toInt.name}()"
+                    }
+                }
```

Comment:
Is it possible to check if cast required in `getOperandExpectedType`? 
And pass correct expected type, to let `CodeConverter.convertExpression` handle type conversions.

Reply:
Thanks @semoro for your patience, was very busy lately.
Let me see what I can do here :)

Reply:
Hey @semoro, I'm not exactly sure what you're requesting, could you help me out?

Reply:
Yep, Now `toInt` added only while converting literal, but it's not sufficient:
```
public class TestArithmWithChars {
    public int foo(char x) {
        return '0' >> x;
    }
}
```
Converted to: 
```
class TestArithmWithChars {
    fun foo(x: Char): Int {
        return '0'.toInt() shr x //.toInt missing
    }
}
```

This problem caused by incorrect type passed to `CodeConverter.convertExpresion`[`CodeConverter.kt:97`](https://github.com/paplorinc/kotlin/blob/e0cf221a18bb9bb6bc33f8df1158ec9592fb9275/j2k/src/org/jetbrains/kotlin/j2k/CodeConverter.kt#L97)

This type incorrectly computed in `ExpressionConverter.visitBinaryExpression` [`ExpressionConverter.kt:113`](https://github.com/paplorinc/kotlin/blob/e0cf221a18bb9bb6bc33f8df1158ec9592fb9275/j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt#L113)

Reply:
Fixed, you were right, the fix should have been in `getOperandExpectedType` instead.
I like this version a lot better than the original, thanks again for the hints @semoro! :)

I pushed the new version, would you like me to rebase the changes so that it doesn't contain the wrong modifications?

Reply:
I'll rebase it myself on merging, thanks! Will review again on days